fe/fe-core/src/main/java/com/starrocks/analysis/InvertedIndexUtil.java

```diff
@@ -0,0 +1,85 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.analysis;
+
+import com.starrocks.catalog.Column;
+import com.starrocks.catalog.KeysType;
+import com.starrocks.catalog.PrimitiveType;
+import com.starrocks.catalog.ScalarType;
+import com.starrocks.common.AnalysisException;
+import com.starrocks.common.InvertedIndexParams.IndexParamsKey;
+import com.starrocks.sql.analyzer.SemanticException;
+
+import java.util.Locale;
+import java.util.Map;
+
+import static com.starrocks.common.InvertedIndexParams.CommonIndexParamKey.IMP_LIB;
+import static com.starrocks.common.InvertedIndexParams.InvertedIndexImpType.CLUCENE;
+
+public class InvertedIndexUtil {
+
+    public static String INVERTED_INDEX_PARSER_KEY = IndexParamsKey.PARSER.name().toLowerCase(Locale.ROOT);
+    public static String INVERTED_INDEX_PARSER_NONE = "none";
+    public static String INVERTED_INDEX_PARSER_STANDARD = "standard";
+    public static String INVERTED_INDEX_PARSER_ENGLISH = "english";
+    public static String INVERTED_INDEX_PARSER_CHINESE = "chinese";
+
+    public static String getInvertedIndexParser(Map<String, String> properties) {
+        String parser = properties == null ? null : properties.get(INVERTED_INDEX_PARSER_KEY);
+        // default is "none" if not set
+        return parser != null ? parser : INVERTED_INDEX_PARSER_NONE;
+    }
+
+    private static boolean validGinColumnType(Column column) {
+        return (column.getType() instanceof ScalarType)
+                && (column.getPrimitiveType() == PrimitiveType.CHAR || column.getPrimitiveType() == PrimitiveType.VARCHAR);
+    }
+
+    public static void checkInvertedIndexValid(Column column, Map<String, String> properties, KeysType keysType) {
+        if (keysType != KeysType.DUP_KEYS) {
+            throw new SemanticException("The inverted index can only be build on DUPLICATE table.");
+        }
+        if (!validGinColumnType(column)) {
+            throw new SemanticException("The inverted index can only be build on column with type of scalar type.");
+        }
+
+        String impLibKey = IMP_LIB.name().toLowerCase(Locale.ROOT);
+        if (properties.containsKey(impLibKey)) {
```

Comment:
I think it may be necessary to check if there are other illegal kv values ​​in the properties.